torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1555,7 +1555,46 @@ void LoopNest::reorderAxis(For* a, For* b) {
   if (after) {
     root->insert_stmt_after(after, newInner);
   }
-} // namespace tensorexpr
+}
+
+std::vector<For*> LoopNest::reorder(
+    const std::vector<For*>& loops,
+    const std::vector<size_t>& permutation) {
+  TORCH_INTERNAL_ASSERT(loops.size() == permutation.size());
+  if (loops.size() < 2) {
+    // Nothing to reorder here.
+    return loops;
+  }
+
+  // Check if the list of loops are perfectly nested.
```

Comment:
As you work on this, could you please make this a utility function? It could be useful in other transformations, such as `rfactor`.

Reply:
Good point. Added.